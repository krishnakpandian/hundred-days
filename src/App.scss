.app {
  text-align: center;
  background-image: url('../src/Assets/background.jpg');
  background-size: cover;
  background-position: center;
  transition: all 2s ease-in 1s;
  -webkit-transition: all 2s ease-in 1s;
  -moz-transition: all 2s ease-in 1s;
  -o-transition: all 2s ease-in 1s;
}
