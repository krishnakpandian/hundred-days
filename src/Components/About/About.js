import React from 'react';
import './About.scss'

const About = () => {
    return(
        <React.Fragment>
            <div class="about-container">
                <div class="title">About Me</div>
                <div class="info">A little bit about me, I am a third year Computer Engineering major at UC Santa Cruz. I am really interested in Software Engineering, networking, IOT, and embedded systems. I created this site from scratch as a way for me to get exposure in a lot of these areas. I will be updating this site daily as a blog and link all the work I do during this time!</div>
                <div class="changes">Slight Changes: I decided I wanted to tweak the rules a little bit from the FreeCodeCamp version but it will mostly be the same. I will be doing daily projects/mini-projects in the areas I am personally interested in opposed to the guided curriculum and I am using my own template for recording all of the updates. Otherwise, I am excited to get this started and see how long we can go!</div>
            </div>
        </React.Fragment>
    );
}

export default About;