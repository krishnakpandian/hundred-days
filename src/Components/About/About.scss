.about-container {
    min-height: 10vh;
    margin: 0 10vw 0 10vw;
    color: azure;
    .title {
        font-size: 22px;
        line-height: 25px;
        margin-bottom: 2vh;
    }
    .info {
        font-size: 18px;
        margin-bottom: 2vh;
    }
    .changes {
        font-size:16px;
        margin-bottom: 2vh;
    } 
}