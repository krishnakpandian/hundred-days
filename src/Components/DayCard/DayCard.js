import React from 'react';
import './DayCard.scss';
import Card from 'react-bootstrap/Card';
import Fade from 'react-reveal/Fade';

const DayCard = (props) => {
    if (props.data.challenge === "Fail") {
        return(
        <React.Fragment>
            <div class="card-container fail">
                <Fade down delay={300} distance={"0.5em"}>
                    <Card bg='warning'>
                        <Card.Body>
                            <Card.Title>Day: {props.index}</Card.Title>
                            <Card.Subtitle className="mb-2 text-muted">{props.data.challenge}</Card.Subtitle>
                            <Card.Text>{props.data.description}</Card.Text>
                            <Card.Link>{props.data.date}</Card.Link>
                        </Card.Body>
                    </Card>
                </Fade>
            </div>
        </React.Fragment>
        );
    }
    else {
        return (
            <React.Fragment>
                <div class="card-container">
                    <Fade down delay={300} distance={"0.5em"}>
                        <Card>
                            <Card.Body>
                                <Card.Title>Day: {props.index}</Card.Title>
                                <Card.Subtitle className="mb-2 text-muted">{props.data.challenge}</Card.Subtitle>
                                <Card.Text>{props.data.description}</Card.Text>
                                <Card.Link href={props.data.link}>{props.data.date}</Card.Link>
                            </Card.Body>
                        </Card>
                    </Fade>
                </div>
            </React.Fragment>
        );
    }
}

export default DayCard;