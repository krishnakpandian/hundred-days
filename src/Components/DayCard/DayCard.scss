.card-container {
    &.fail {
        background-color: red;
        color: white;
    }
    opacity: 1;
    transition: 500ms;
    overflow: none;
    color:blue;
    background-color: azure;
    .card {
        width: 23vw;
        height: 35vh;
    }   

}

@media screen and (max-width: 1070px) {
    .card-container {
        .card {
            width: 40vw;
            height: 40vw;
            .card-body {
                .card-text {
                    font-size: 13px;
                }
            }
        }
    }
}


@media screen and (max-width: 768px) {
    .card-container {
        .card {
            .card-body {
                .card-text {
                    font-size: 16px;
                }
            }
        }   
    }
}