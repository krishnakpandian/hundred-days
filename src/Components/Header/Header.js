import React from 'react';
import './Header.scss';

const Header = () => {
    return(
        <React.Fragment>
            <div class="header-container">
                <header>
                    Krishna's 100 Day of Code Challenge
                </header>
            </div>
        </React.Fragment>
    );
}

export default Header;