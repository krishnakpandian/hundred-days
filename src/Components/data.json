{
    "data": [
        {
            "date": "August 9, 2020",
            "challenge": "Starting 100 Days of Code Challenge",
            "description": "I will be starting my 100 days of code today and for day 1 and most of week 1, I will be setting up the react App to get this ready, Links to all commits will be here ",
            "link": "https://github.com/krishnakpandian/hundred-days/commit/0e2271e062e0d8bdfa99d401c079a6318054a614"
        },
        {
            "date": "August 10, 2020",
            "challenge": "Get the Design of this Site Working",
            "description": "Today I will be working with ReactJS, HTML, CSS, to get most of the starter material working for this challenge.",
            "link": "https://github.com/krishnakpandian/hundred-days/commit/505869cf9245ea260fc481e11ae0191d390b3709"
        },
        {
            "date": "August 11, 2020",
            "challenge": "Finishing the Website and Deployment",
            "description": "This site is now live on Firebase and it should be visible to everyone now!",
            "link": "https://hundreddaychallenge-fe11d.firebaseapp.com/"
        },
        {
            "date": "August 12, 2020",
            "challenge": "Unit Testing Our Website using Jest and Enzyme and more refining touches to the frontend",
            "description": "By the end of this week, I want to have this site fully ready, today we clean up the cards a little more and learn how to unit test using Jest and Enzyme",
            "link": ""
        },
        {
            "date": "August 13, 2020",
            "challenge": "",
            "description": "",
            "link": ""
        }
    ]
}