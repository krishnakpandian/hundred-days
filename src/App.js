import React from 'react';
import './App.scss';
import Footer from './Components/Footer/Footer.js';
import Blog from './Components/Blog/Blog.js';
import About from './Components/About/About.js';
import Header from './Components/Header/Header.js';
import Closure from './Components/Closure/Closure.js';
const App = () => {
  return (
    <>
    <div class="app">
      <Header/>
      <About/>
      <Blog/>
      <Closure/>
      <Footer/>
    </div>
    </>
  );
}

export default App;
