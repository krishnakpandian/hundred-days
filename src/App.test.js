import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import Header from './Components/Header/Header.js';
import Footer from './Components/Footer/Footer.js';
import About from './Components/About/About.js';
import DayCard from './Components/DayCard/DayCard.js';
import { mount, shallow, render as testing } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import data from './Components/data.json';
Enzyme.configure({ adapter: new Adapter() });

test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

describe('Header renders', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Header debug />);
  
    expect(component).toMatchSnapshot();
  });
})

describe('Footer renders', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Footer debug />);
  
    expect(component).toMatchSnapshot();
  });
})

describe('About renders', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<About debug />);
  
    expect(component).toMatchSnapshot();
  });
})


describe('DayCard renders Correctly', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<DayCard data={data.data[0]} index= {0} />);
  
    expect(component).toMatchSnapshot();
  });
})

describe('DayCard renders Incorrectly', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<DayCard debug />);
  
    expect(component).toMatchSnapshot();
  });
})