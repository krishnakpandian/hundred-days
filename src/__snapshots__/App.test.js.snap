// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`DayCard renders Correctly should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`Footer renders should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`Header renders should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
